### DISCUSSION TITLE
chown -R apache not working in macOS

### DISCUSSION THREAD

**Discussion by icetechy** [laravel/framework#40764] [General] [answered]
How to give permission to Laravel folders to be written by apache and Laravel.

I just install Laravel-8 on my mac so I am trying to run the below commands via Terminal on Mac:

*These commands: chmod -R 775 storage/

chmod -R 775 bootstrap/cache

worked perfectly but the below commands didn't work:

chown -R apache storage/ and chown -R apache bootstrap/cache

I keep getting this error "chown: apache: illegal user name".

Please, how do I get this issue resolved?

**Comment by chuckrincon** [comment, marked as answer]
Hey there ✋🏻 .

Basically the `apache` group doesn't exist, you can create it and then run again the command.

Try using just the owner if no group
```bash
chown -R $USER storage
chown -R $USER bootstrap/cache
```

But I highly recommend you to use the Laravel's default server environment, by running this command:
```bash
php artisan serve
```

If you have [Brew](https://brew.sh/) installed in your machine, the best way to prevent and avoid this kind of issues is using [Laravel Valet](https://laravel.com/docs/8.x/valet), which is a development environment for macOS.

Or if you are using Docker, there is [Laravel Sail](https://github.com/laravel/sail), a light-weight command-line interface for interacting with Laravel's default Docker development environment.

Using any of the options above, prevent and avoid this kind of issues.

**Reply by icetechy** [reply]
Thank you very much, this worked like a charm :-)

**Comment by ousid** [comment]
Hey, @icetechy 
I created a [blog post](https://coderflex.com/blog/2-ways-to-setup-file-permissions-for-laravel-in-production) about permission on Laravel months ago, give it a read, you may find the solution in there.
